File: 948-sort-an-array/sort-an-array.py
class Solution:
    def merge(self,nums,lo,mid,hi):
        i=lo
        j=mid+1
        temp=[]
        while i<=mid and j<=hi:
            if nums[i]<=nums[j]:
                temp.append(nums[i])
                i=i+1
            else:
                temp.append(nums[j])
                j=j+1
        while i<=mid:
            temp.append(nums[i])
            i=i+1
        while j<=hi:
            temp.append(nums[j])
            j=j+1
        for i in range(lo,hi+1):
            nums[i]=temp[i-lo]

    def mergeSort(self,nums,lo,hi):
        if lo>=hi:
            return 
        mid=(lo+hi)//2
        self.mergeSort(nums,lo,mid)
        self.mergeSort(nums,mid+1,hi)
        self.merge(nums,lo,mid,hi)
    def sortArray(self, nums: List[int]) -> List[int]:
        self.mergeSort(nums,0,len(nums)-1)
        return nums

File: 948-sort-an-array/sort-an-array.java
class Solution {
     public void merge(int low, int mid,int high,int []nums){
        int i=low;
        int j=mid+1;
        ArrayList<Integer> temp=new ArrayList<>();

        while(i<=mid && j<=high){
            if(nums[i]<nums[j]){
                temp.add(nums[i]);
                i++;
            }else{
                temp.add(nums[j]);
                j++;
            }
        }

        while(i<=mid){
            temp.add(nums[i]);
            i++;
        }

        while(j<=high){
            temp.add(nums[j]);
            j++;
        }

        for(int k=low;k<=high;k++){
            nums[k]=temp.get(k-low);
        }
    }

    public void mergeSorted(int low,int high, int []nums){
        if(low>=high){
            return ;
        }
        int mid=(low+high)/2;
        mergeSorted(low,mid,nums);
        mergeSorted(mid+1,high,nums);
        merge(low,mid,high,nums);

    }
    public int[] sortArray(int[] nums) {

        mergeSorted(0,nums.length-1,nums);
        return nums;
        
    }
}